### DISCUSSION TITLE
Code blocks in footnotes are not aligned with the footnote

### DISCUSSION THREAD

**Discussion by asmeurer** [pradyunsg/furo#729] [Potential Issue]
Example: 
<img width="761" alt="image" src="https://github.com/pradyunsg/furo/assets/71486/80be67f4-ef12-4c7f-ab11-ab205d6b6a29">

Generated from

```rst
Some text with a footnote [1]_



.. [1]

   Footnote

   .. code::

      test code

   More footnote

```

I would expect the code block to be indented to align with "footnote". This looks bad if you have several footnotes with code blocks.

I'm not completely sure that this is a Furo issue, because I checked and the same thing with "alabaster" and "classic" and they both have the same issue. However, from what I can tell, the HTML seems well formed enough, so it's not clear to me that the problem is with what Sphinx is generating.

**Comment by saurabheights** [comment]
This is probably similar to https://github.com/readthedocs/sphinx_rtd_theme/issues/1540 where issue is in citation references. Changing theme to furo produces similar issue.

**Reply by saurabheights** [reply]
P.S. The issue is coming from using `docutils==0.18.1`.

**Reply by pradyunsg** [reply]
Thanks for exploring this and figuring that out @saurabheights! ^>^

**Reply by saurabheights** [reply]
Forgot to respond with complete information earlier. `docutils==0.17.0` works fine, it probably broke somewhere at version `0.18.x`.

You can use code from https://github.com/readthedocs/sphinx_rtd_theme/issues/1540 to reproduce the issue. 

In below image, you can see that last two references lead to very different html output. Also, issue is not just for last reference, but happens before with `ArujoAndOliveira2020` reference too.

![image](https://github.com/pradyunsg/furo/assets/11634769/d2d4f4fc-04ab-4123-bdf6-5d71efa6937e)
